### PR TITLE
Feature/Add address and date time in SendLocation RPC

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Send location now support new fields.

CRQ details: [SendLocation: SDL must support new "address" and "timeStamp" parameters](https://adc.luxoft.com/jira/browse/APPLINK-23322)

@AByzhynar @anosach-luxoft @OHerasym @dev-gh please review

### CODE REVIEW

**Comment by AByzhynar** [comment]
Reviewed

File: src/components/application_manager/src/commands/mobile/send_location_request.cc [hunk 1]
```diff
@@ -60,7 +62,7 @@ void SendLocationRequest::Run() {
     return;
   }
 
-  const smart_objects::SmartObject& msg_params = (*message_)[strings::msg_params];
+  SmartObject& msg_params = (*message_)[strings::msg_params];
```
Comment (dev-gh) [inline src/components/application_manager/src/commands/mobile/send_location_request.cc]: Why you've removed constness?

File: src/components/application_manager/src/commands/mobile/send_location_request.cc [hunk 2]
```diff
@@ -99,59 +107,77 @@ void SendLocationRequest::Run() {
...
+  const smart_objects::SmartObject& msg_params = (*message_)[strings::msg_params];
+  MessageHelper::PrintSmartObject(msg_params);
+  const bool longitude_degrees_exist =
+      msg_params.keyExists(strings::longitude_degrees);
+  const bool latitude_degrees_exist =
+      msg_params.keyExists(strings::latitude_degrees);
+  const bool address_exist = msg_params.keyExists(strings::address);
+
+  if (latitude_degrees_exist ^ longitude_degrees_exist) {
+    LOG4CXX_DEBUG(logger_, "latitudeand longitude should be provided only in pair");
```
Comment (dev-gh) [inline src/components/application_manager/src/commands/mobile/send_location_request.cc]: typo

File: src/components/application_manager/src/commands/mobile/send_location_request.cc (hunk 1 above)
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/send_location_request.cc]: @dev-gh  before my changes for functions that was need not const msg_params used *(message_)[strings::msg_params], I thins that not const variable is better that using hack for hiding usage of not const msg_params

**Comment by LuxoftAKutsan** [comment]
@TYasyniuk Please review API changes in commit  	404b7fa

File: src/components/application_manager/src/commands/mobile/send_location_request.cc (hunk 2 above)
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/send_location_request.cc]: @dev-gh  fixed 1a3b88d

**Comment by TMelnyk** [comment]
reviewed

**Comment by dev-gh** [comment]
Reviewed

**Comment by TMelnyk** [comment]
Alexandr, please change the type of "longtitudeDegrees" and "latitudeDegrees" parameters to **Double ** per MOBILE_API

**Comment by LuxoftAKutsan** [comment]
Squashed all commits, fixed Float -> Double data type.
Do not merge until PASA confirm.

**Comment by AGaliuzov** [comment]
@LuxoftAKutsan how does the PASA affect this code? Why we need to wait?

**Comment by LuxoftAKutsan** [comment]
@AGaliuzov this question is not clear for me too. You can try to find arguments with NSnitsar.

**Comment by AGaliuzov** [comment]
@LuxoftAKutsan anything else so far, or I can merge this PR?

**Comment by LuxoftAKutsan** [comment]
@AGaliuzov  please merge

**Pull request merged**